bird_cloud_gnn/radar_dataset.py

```diff
@@ -152,19 +163,32 @@ def process(self):
 
         self.graphs = []
         self.labels = np.array([])
-        for data_file in os.listdir(self.data_folder):
-            self._read_one_file(data_file)
+        if self.data_folder is not None:
+            if os.path.isdir(self.data_folder):
+                for data_file in os.listdir(self.data_folder):
+                    self._read_one_file(os.path.join(self.data_folder, data_file))
+            else:
+                if not os.path.isfile(self.data_folder):
+                    raise ValueError("`data_folder` is neither a file nor a directory")
+                self._read_one_file(self.data_folder)
```

Comment:
If we make the check dir vs file on __init__, then this can be simplified. After we decide how, we can come back to this.

Reply:
Coming back to this.
Flattening makes it easier to read and reason:

```suggestion
        if self.data_folder is not None:
            if os.path.isdir(self.data_folder):
                for data_file in os.listdir(self.data_folder):
                    self._read_one_file(os.path.join(self.data_folder, data_file))
            elif os.path.isfile(self.data_folder):
                self._read_one_file(self.data_folder)
            else:
                raise ValueError("`data_folder` is neither a file nor a directory")
```